vehicle/saic/requests/sendRequest.go

```diff
@@ -0,0 +1,100 @@
+package requests
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func SendRequest(
```

Comment:
If we need this for all requests, a transport decorator would be nice to use.

Reply:
I refactored the communication. Now some of the headers are added by a decorator.
Not sure that this is what you intended. Only the constant headers can be added by a decorator. All the dynamic ones are also used in the body (or for the encryption) so it doesn't make sense to delegate that to a decorator.
However, I tried to stick to the already implemented logic of using local transports for the API and the identity.

Reply:
> Only the constant headers can be added by a decorator.

oauth2 decorates the token. We could do the same and make the entire request encryption a decorator. That may be more a topic of "nicer code" though.

Reply:
Ok. Didn't know that decorators can be used to modify the body of a rquest.